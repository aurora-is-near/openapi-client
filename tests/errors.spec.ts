import {
  isOpenApiClientError,
  OpenApiClientError,
  isOpenApiClientTimeoutError,
  OpenApiClientTimeoutError,
  OpenApiClientTimeoutErrorCode,
} from '../src/errors';

describe('Errors', () => {
  describe('OpenApiClientError', () => {
    it('creates a basic error', () => {
      const err = new OpenApiClientError(400, 'BadRequest');

      expect(err.statusCode).toBe(400);
      expect(err.type).toBeUndefined();
      expect(err.errors).toBeUndefined();
      expect(err.message).toMatchSnapshot();
    });

    it('creates an error with a type and validation errors', () => {
      const err = new OpenApiClientError(400, 'BadRequest', {
        type: '/probs/bad-request',
        errors: [
          {
            property: 'email',
            message: 'The email is no good',
            constraint: 'isValid',
          },
        ],
      });

      expect(err.statusCode).toBe(400);
      expect(err.type).toBe('/probs/bad-request');
      expect(err.errors).toEqual([
        {
          property: 'email',
          message: 'The email is no good',
          constraint: 'isValid',
        },
      ]);

      expect(err.message).toMatchSnapshot();
    });
  });

  describe('isOpenApiClientError', () => {
    it('returns true if an error is a OpenApiClientError', () => {
      const error = new OpenApiClientError(500, 'Internal Server Error');

      expect(isOpenApiClientError(error)).toBe(true);
    });

    it('returns false if an error is not a OpenApiClientError', () => {
      const error = new Error();

      expect(isOpenApiClientError(error)).toBe(false);
    });

    it('returns false if an error is not an error object', () => {
      expect(isOpenApiClientError('')).toBe(false);
    });
  });

  describe('OpenApiClientTimeoutError', () => {
    it.each(['ECONNABORTED', 'ETIMEDOUT'])(
      'creates a timeout error for status code %s',
      (code) => {
        const err = new OpenApiClientTimeoutError(
          'Request aborted',
          code as OpenApiClientTimeoutErrorCode,
        );

        expect(err.message).toBe('Request aborted');
        expect(err.code).toBe(code);
        expect(err.message).toMatchSnapshot();
      },
    );
  });

  describe('isOpenApiClientTimeoutError', () => {
    it.each(['ECONNABORTED', 'ETIMEDOUT'])(
      'returns true if an error is an OpenApiClientTimeoutError and %s',
      (code) => {
        const error = new OpenApiClientTimeoutError(
          'Request aborted',
          code as OpenApiClientTimeoutErrorCode,
        );

        expect(isOpenApiClientTimeoutError(error)).toBe(true);
      },
    );

    it('returns false if an error is not an OpenApiClientTimeoutError', () => {
      const error = new Error();

      expect(isOpenApiClientTimeoutError(error)).toBe(false);
    });

    it('returns false if an error is not an error object', () => {
      expect(isOpenApiClientTimeoutError('')).toBe(false);
    });
  });
});
