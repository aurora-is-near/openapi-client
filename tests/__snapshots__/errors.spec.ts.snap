// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors OpenApiClientError creates a basic error 1`] = `"BadRequest"`;

exports[`Errors OpenApiClientError creates an error with a type and validation errors 1`] = `"BadRequest The email is no good (email isValid)"`;

exports[`Errors OpenApiClientTimeoutError creates a timeout error for status code ECONNABORTED 1`] = `"Request aborted"`;

exports[`Errors OpenApiClientTimeoutError creates a timeout error for status code ETIMEDOUT 1`] = `"Request aborted"`;
